src/SFA.DAS.AssessorService.Application.Api/Controllers/Validations/WithdrawalDateValidationController.cs

```diff
@@ -0,0 +1,62 @@
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Api.Types.Models.Apply;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.Api.Controllers.Validations
+{
+    [Authorize(Roles = "AssessorServiceInternalAPI")]
+    public class WithdrawalDateValidationController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly ILogger<WithdrawalDateValidationController> _logger;
+
+        public WithdrawalDateValidationController(IQnaApiClient qnaApiClient, IMediator mediator, ILogger<WithdrawalDateValidationController> logger)
+        {
+            _qnaApiClient = qnaApiClient;
+            _mediator = mediator;
+            _logger = logger;
+        }
+        
+        [HttpGet("Validations/WithdrawalDate/{id}")]
+        public async Task<ActionResult<ApiValidationResult>> ValidateWithdrawalDate(Guid id, string q)
+        {
+            try
+            {
+                var application = await _mediator.Send(new GetApplicationRequest(id));
+                var appData = await _qnaApiClient.GetApplicationData(application.ApplicationId);
+
+                var dateComponents = q.Split(",");
+                var enteredDate = new DateTime(int.Parse(dateComponents[2]), int.Parse(dateComponents[1]), int.Parse(dateComponents[0]));
+                if (enteredDate < appData.EarliestDateOfWithdrawal.Date)
```

Comment:
The earliest date of withdrawal that's stored with the application data. How is that calculated? Is that something we can get here a different way? It feels odd to have to call the QnA Api and setup all the config just to get that one thing if we can replicate the calculation. If it's not possible, happy with what's here, just wanted to understand the logic under the hood QnA side

Reply:
Removed the qna api call, now using the method  that calculates the earliestDateOfWithdrawal